### PR TITLE
Test change managing entity

### PR DESCRIPTION
**Pull request by Raynafs**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes [link to issue]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt [hunk 1]
```diff
@@ -200,10 +203,33 @@ class ProfileViewModelTest : RobolectricTest() {
...
+        visible = "true",
+        showSeparator = false,
+        enabled = "true",
+        actions = emptyList(),
+      )
+    val navController = mockk<NavController>()
+    val event = ProfileEvent.OverflowMenuClick(navController, resourceData, overflowMenuItemConfig)
+    val mockManagingEntity: ManagingEntityConfig? = null
+    profileViewModel.changeManagingEntity(event, mockManagingEntity)
+    verify { Timber.w("ManagingEntityConfig required. Base resource should be Group") }
```
Comment (ellykits) [inline android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt:233]: Can we add extra verification/assertion here other than just verifying the log? the log here is subject to change. Verify/assert something that affects the `profileViewModel.changeManagingEntity` functionality instead.

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt (hunk 1 above)
Comment (Raynafs) [reply android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt:233]: Yes sure working on it

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt [hunk 2]
```diff
@@ -200,10 +201,25 @@ class ProfileViewModelTest : RobolectricTest() {
...
     every { mockProfileViewModel.onEvent(any()) } just runs
-
     verifyAll {
       mockActionConfig.interpolate(any())
       profileViewModel.onEvent(event)
     }
   }
+
+  @Test
+  fun testThatChangeManagingEntityDisplaysBottomSheetWhenEligibleEntitiesExist() {
```
Comment (FikriMilano) [inline android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt:211]:
```suggestion
  fun testChangeManagingEntityDisplaysBottomSheetWhenResourceTypeIsGroup() {
```

Just an example, feel free to re-word it, for better clarity on what makes it eligible.

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt [hunk 3]
```diff
@@ -200,10 +201,25 @@ class ProfileViewModelTest : RobolectricTest() {
...
+
+    every { event.resourceData?.baseResourceType } returns ResourceType.Group
+    coEvery { registerRepository.loadResource<Group>(any()) } returns mockk(relaxed = true)
+    every { fhirPathDataExtractor.extractValue(any(), any()) } returns "true"
+
+    profileViewModel.changeManagingEntity(event, managingEntity)
+
+    coVerify { registerRepository.loadResource<Group>(any()) }
+    coVerify { fhirPathDataExtractor.extractValue(any(), any()) }
+    coVerify(exactly = 0) { profileViewModel.emitSnackBarState(any()) }
```
Comment (FikriMilano) [inline android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt:223]: Might be difficult, but is it possible to also check the bottom sheet? to ensure its shown.

If it is, we can skip that.

**Review by FikriMilano — COMMENTED** [review]
Thanks! Just some comments

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt (hunk 2 above)
Comment (Raynafs) [reply android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt:211]: Yes sure working on it

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt (hunk 3 above)
Comment (Raynafs) [reply android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt:223]: Let me give it a shot

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt [hunk 4]
```diff
@@ -200,10 +201,25 @@ class ProfileViewModelTest : RobolectricTest() {
...
+
+  @Test
+  fun testThatChangeManagingEntityDisplaysBottomSheetWhenEligibleEntitiesExist() {
+    val event = mockk<ProfileEvent.OverflowMenuClick>(relaxed = true)
+    val managingEntity: ManagingEntityConfig = mockk(relaxed = true)
+
+    every { event.resourceData?.baseResourceType } returns ResourceType.Group
+    coEvery { registerRepository.loadResource<Group>(any()) } returns mockk(relaxed = true)
+    every { fhirPathDataExtractor.extractValue(any(), any()) } returns "true"
+
```
Comment (ellykits) [inline android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt:218]: Would it be possible NOT to use mock? The FhirPathDataExtractor for instance is just a utility class for extracting data from a Resource, mocking it is unnecessary.

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt (hunk 4 above)
Comment (Raynafs) [reply android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt:218]: Oh alright. working on it

**Pull request closed without merging**